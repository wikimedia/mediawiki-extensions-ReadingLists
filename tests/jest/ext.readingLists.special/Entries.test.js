const { mount } = require( '@vue/test-utils' );
const api = require( '../../../resources/ext.readingLists.api/index.js' );

const LIST = require( '../fixtures/list.json' );
const ENTRIES = require( '../fixtures/entries.json' );
const PAGES = require( '../fixtures/pages.json' );

function setupEntriesApiStub() {
	return api.stubApi( {
		get: jest.fn( ( { action, meta, rllist, list, rlelists, prop } ) => {
			if ( action === 'query' ) {
				if ( meta === 'readinglists' && rllist === 12345 ) {
					return LIST;
				} else if ( list === 'readinglistentries' && rlelists === 12345 ) {
					return ENTRIES;
				} else if ( prop !== undefined ) {
					return PAGES;
				}
			}
		} )
	} );
}

describe( 'Entries', () => {
	test( 'renders with toolbar disabled', async () => {
		setupEntriesApiStub();
		mw.util = { getUrl: jest.fn( ( path ) => `/wiki/${ path }` ) };

		const Entries = require( '../../../resources/ext.readingLists.special/pages/Entries.vue' );
		const wrapper = mount( Entries, { props: { listId: 12345 } } );

		Object.defineProperty( wrapper.vm, 'enableToolbar', { value: false, writable: true } );

		await wrapper.vm.$forceUpdate();
		await wrapper.vm.$nextTick();

		expect( wrapper.element ).toMatchSnapshot();
	} );

	test( 'renders with toolbar enabled', async () => {
		setupEntriesApiStub();
		mw.util = { getUrl: jest.fn( ( path ) => `/wiki/${ path }` ) };

		const Entries = require( '../../../resources/ext.readingLists.special/pages/Entries.vue' );
		const wrapper = mount( Entries, { props: { listId: 12345 } } );

		Object.defineProperty( wrapper.vm, 'enableToolbar', { value: true, writable: true } );

		wrapper.vm.ready = true;
		wrapper.vm.loadingEntries = true;

		await wrapper.vm.$forceUpdate();
		await wrapper.vm.$nextTick();

		expect( wrapper.element ).toMatchSnapshot();
	} );
} );
