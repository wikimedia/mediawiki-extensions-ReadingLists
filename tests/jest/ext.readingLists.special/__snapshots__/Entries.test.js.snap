// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entries renders with toolbar disabled 1`] = `
<div
  data-v-app=""
>
  
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  <div>
     (sorted by most recent) 
  </div>
  <!--v-if-->
  <!--v-if-->
  <div
    aria-disabled="false"
    aria-label="Please wait..."
    class="cdx-progress-bar cdx-progress-bar--block cdx-progress-bar--enabled"
    role="progressbar"
  >
    <div
      class="cdx-progress-bar__bar"
    />
  </div>
  
</div>
`;

exports[`Entries renders with toolbar enabled 1`] = `
<div
  data-v-app=""
>
  
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  <div
    class="reading-lists-toolbar"
  >
    <div
      class="cdx-menu-button"
    >
      <button
        aria-controls="v-1"
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="Display menu"
        aria-pressed="false"
        class="cdx-toggle-button cdx-toggle-button--quiet cdx-toggle-button--toggled-off cdx-toggle-button--icon-only"
        disabled=""
        id="v-0"
        type="button"
      >
        
        
        <span
          class="cdx-icon cdx-icon--medium"
        >
          <svg
            aria-hidden="true"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <!--v-if-->
            <g>
              <path
                d="M10 0 3 8h14zm0 18-7-8h14z"
              />
            </g>
          </svg>
        </span>
        
        
      </button>
      <div
        class="cdx-menu-button__menu-wrapper"
      >
        <div
          class="cdx-menu"
          style="display: none; visibility: hidden; position: absolute; top: 0px; left: 0px; transform: translate(0px, 0px);"
        >
          <ul
            aria-labelledby="v-0"
            aria-multiselectable="true"
            class="cdx-menu__listbox"
            id="v-1"
            role="menu"
            tabindex="-1"
          >
            <!--v-if-->
            <!--v-if-->
            
            
            <li
              class="cdx-menu__group-wrapper"
            >
              <ul
                aria-describedby="v-2-1-description"
                aria-labelledby="v-2-1-label"
                class="cdx-menu__group"
                role="group"
              >
                <span
                  class="cdx-menu__group__meta"
                >
                  <!--v-if-->
                  <span
                    class="cdx-menu__group__meta__text"
                  >
                    <span
                      class="cdx-menu__group__label"
                      id="v-2-1-label"
                    >
                      Sort by
                    </span>
                    <!--v-if-->
                  </span>
                </span>
                
                <li
                  aria-checked="true"
                  aria-disabled="false"
                  class="cdx-menu-item cdx-menu-item--selected cdx-menu-item--enabled cdx-menu__group__item"
                  id="v-2-2"
                  role="option"
                >
                  
                  <span
                    class="cdx-menu-item__content"
                  >
                    <span
                      class="cdx-icon cdx-icon--medium cdx-menu-item__icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="M17.66 18h-2a.85.85 0 01-.56-.17 1.1 1.1 0 01-.32-.43l-1.33-3.53h-6.9L5.22 17.4a1.06 1.06 0 01-.31.41.83.83 0 01-.56.19h-2L8.68 2h2.63zm-4.92-6-2.2-5.84A16 16 0 0110 4.43q-.12.52-.27 1t-.27.77L7.26 12z"
                          />
                        </g>
                      </svg>
                    </span>
                    <span
                      class="cdx-menu-item__text"
                    >
                      <span
                        class="cdx-menu-item__text__label"
                      >
                        <bdi>
                          Name
                        </bdi>
                      </span>
                      <!--v-if-->
                      <!--v-if-->
                      <!--v-if-->
                    </span>
                    <span
                      class="cdx-icon cdx-icon--small cdx-menu-item__selected-icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="M7 14.17 2.83 10l-1.41 1.41L7 17 19 5l-1.41-1.42z"
                          />
                        </g>
                      </svg>
                    </span>
                  </span>
                  
                </li>
                <li
                  aria-disabled="false"
                  class="cdx-menu-item cdx-menu-item--enabled cdx-menu__group__item"
                  id="v-2-3"
                  role="option"
                >
                  
                  <span
                    class="cdx-menu-item__content"
                  >
                    <span
                      class="cdx-icon cdx-icon--medium cdx-menu-item__icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="M9 6v5h.06l2.48 2.47 1.41-1.41L11 10.11V6z"
                          />
                          <path
                            d="M10 1a9 9 0 00-7.85 13.35L.5 16H6v-5.5l-2.38 2.38A7 7 0 1110 17v2a9 9 0 000-18"
                          />
                        </g>
                      </svg>
                    </span>
                    <span
                      class="cdx-menu-item__text"
                    >
                      <span
                        class="cdx-menu-item__text__label"
                      >
                        <bdi>
                          Updated
                        </bdi>
                      </span>
                      <!--v-if-->
                      <!--v-if-->
                      <!--v-if-->
                    </span>
                    <!--v-if-->
                  </span>
                  
                </li>
                
              </ul>
            </li>
            
            
            <li
              class="cdx-menu__group-wrapper"
            >
              <ul
                aria-describedby="v-2-4-description"
                aria-labelledby="v-2-4-label"
                class="cdx-menu__group"
                role="group"
              >
                <span
                  class="cdx-menu__group__meta"
                >
                  <!--v-if-->
                  <span
                    class="cdx-menu__group__meta__text"
                  >
                    <span
                      class="cdx-menu__group__label"
                      id="v-2-4-label"
                    >
                      Direction
                    </span>
                    <!--v-if-->
                  </span>
                </span>
                
                <li
                  aria-checked="true"
                  aria-disabled="false"
                  class="cdx-menu-item cdx-menu-item--selected cdx-menu-item--enabled cdx-menu__group__item"
                  id="v-2-5"
                  role="option"
                >
                  
                  <span
                    class="cdx-menu-item__content"
                  >
                    <span
                      class="cdx-icon cdx-icon--medium cdx-menu-item__icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="m2 10 1.42 1.41L9 5.83V18h2V5.83l5.59 5.58L18 10l-8-8z"
                          />
                        </g>
                      </svg>
                    </span>
                    <span
                      class="cdx-menu-item__text"
                    >
                      <span
                        class="cdx-menu-item__text__label"
                      >
                        <bdi>
                          Ascending
                        </bdi>
                      </span>
                      <!--v-if-->
                      <!--v-if-->
                      <!--v-if-->
                    </span>
                    <span
                      class="cdx-icon cdx-icon--small cdx-menu-item__selected-icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="M7 14.17 2.83 10l-1.41 1.41L7 17 19 5l-1.41-1.42z"
                          />
                        </g>
                      </svg>
                    </span>
                  </span>
                  
                </li>
                <li
                  aria-disabled="false"
                  class="cdx-menu-item cdx-menu-item--enabled cdx-menu__group__item"
                  id="v-2-6"
                  role="option"
                >
                  
                  <span
                    class="cdx-menu-item__content"
                  >
                    <span
                      class="cdx-icon cdx-icon--medium cdx-menu-item__icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="m2 10 1.42-1.41L9 14.17V2h2v12.17l5.59-5.58L18 10l-8 8z"
                          />
                        </g>
                      </svg>
                    </span>
                    <span
                      class="cdx-menu-item__text"
                    >
                      <span
                        class="cdx-menu-item__text__label"
                      >
                        <bdi>
                          Descending
                        </bdi>
                      </span>
                      <!--v-if-->
                      <!--v-if-->
                      <!--v-if-->
                    </span>
                    <!--v-if-->
                  </span>
                  
                </li>
                
              </ul>
            </li>
            
            
            <li
              class="cdx-menu__group-wrapper"
            >
              <ul
                aria-describedby="v-2-7-description"
                aria-labelledby="v-2-7-label"
                class="cdx-menu__group"
                role="group"
              >
                <span
                  class="cdx-menu__group__meta"
                >
                  <!--v-if-->
                  <span
                    class="cdx-menu__group__meta__text"
                  >
                    <span
                      class="cdx-menu__group__label"
                      id="v-2-7-label"
                    >
                      View as
                    </span>
                    <!--v-if-->
                  </span>
                </span>
                
                <li
                  aria-checked="true"
                  aria-disabled="false"
                  class="cdx-menu-item cdx-menu-item--selected cdx-menu-item--enabled cdx-menu__group__item"
                  id="v-2-8"
                  role="option"
                >
                  
                  <span
                    class="cdx-menu-item__content"
                  >
                    <span
                      class="cdx-icon cdx-icon--medium cdx-menu-item__icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="M2 2h4v4H2zm12 0h4v4h-4zM8 2h4v4H8zM2 14h4v4H2zm12 0h4v4h-4zm-6 0h4v4H8zM2 8h4v4H2zm12 0h4v4h-4zM8 8h4v4H8z"
                          />
                        </g>
                      </svg>
                    </span>
                    <span
                      class="cdx-menu-item__text"
                    >
                      <span
                        class="cdx-menu-item__text__label"
                      >
                        <bdi>
                          Grid
                        </bdi>
                      </span>
                      <!--v-if-->
                      <!--v-if-->
                      <!--v-if-->
                    </span>
                    <span
                      class="cdx-icon cdx-icon--small cdx-menu-item__selected-icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="M7 14.17 2.83 10l-1.41 1.41L7 17 19 5l-1.41-1.42z"
                          />
                        </g>
                      </svg>
                    </span>
                  </span>
                  
                </li>
                <li
                  aria-disabled="false"
                  class="cdx-menu-item cdx-menu-item--enabled cdx-menu__group__item"
                  id="v-2-9"
                  role="option"
                >
                  
                  <span
                    class="cdx-menu-item__content"
                  >
                    <span
                      class="cdx-icon cdx-icon--medium cdx-menu-item__icon"
                    >
                      <svg
                        aria-hidden="true"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <!--v-if-->
                        <g>
                          <path
                            d="M8 6h9v2H8zm0-3h11v2H8zM1 3h6v6H1zm7 11h9v2H8zm0-3h11v2H8zm-7 0h6v6H1z"
                          />
                        </g>
                      </svg>
                    </span>
                    <span
                      class="cdx-menu-item__text"
                    >
                      <span
                        class="cdx-menu-item__text__label"
                      >
                        <bdi>
                          Compact
                        </bdi>
                      </span>
                      <!--v-if-->
                      <!--v-if-->
                      <!--v-if-->
                    </span>
                    <!--v-if-->
                  </span>
                  
                </li>
                
              </ul>
            </li>
            
            
            <!--v-if-->
          </ul>
        </div>
      </div>
    </div>
    <button
      aria-label="Edit articles"
      aria-pressed="false"
      class="cdx-toggle-button cdx-toggle-button--quiet cdx-toggle-button--toggled-off cdx-toggle-button--icon-only"
      disabled=""
      type="button"
    >
      
      <span
        class="cdx-icon cdx-icon--medium"
      >
        <svg
          aria-hidden="true"
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <!--v-if-->
          <g>
            <path
              d="m16.77 8 1.94-2a1 1 0 000-1.41l-3.34-3.3a1 1 0 00-1.41 0L12 3.23zM1 14.25V19h4.75l9.96-9.96-4.75-4.75z"
            />
          </g>
        </svg>
      </span>
      
    </button>
    <!--v-if-->
  </div>
  <!--v-if-->
  <div
    aria-disabled="false"
    aria-label="Please wait..."
    class="cdx-progress-bar cdx-progress-bar--block cdx-progress-bar--enabled"
    role="progressbar"
  >
    <div
      class="cdx-progress-bar__bar"
    />
  </div>
  
</div>
`;
